### DISCUSSION TITLE
Issues with relocate

### DISCUSSION THREAD

**Discussion by daveceerocker** [rshea0/enjinn#22] [Help] [answered]
Hi,

I think we had crossed lines. the m.db was from the D drive that i set up for Engine, not C drive. im not sure if that was the problem.

anyways...i replaced the m.db to the D drive Engine. Ive deleted another track from the folder and then went to enjinn to relocate but when i type the command it only searches in C drive, i cant see how i can change the search to D (D:\Engine Library\TUNES 1)

the reason for D drive is because on my home PC all the tunes are in my D drive because theres no room on my C. I was going to merge the partitions and i know i can do this without losing data, however the moment i do this, the entire engine catalogue will go red because its trying to locate D drive. this is why im running the tests on my work pc. is this making sense? :)

cheers

![wrong path](https://user-images.githubusercontent.com/84126173/118105001-c0db4900-b3d3-11eb-97db-24542dd03678.png)

**Comment by shayded-exe** [comment]
I just released a fix for this earlier today in [v1.2.1](/rshea0/enjinn/releases/v1.2.1).

Run `enjinn update` and try again.

It'll ask you for a path to search in. You should be able to enter `D:\` and it'll search that drive.

Thanks for opening a topic on here btw. It helps me keep track of everything.

**Reply by daveceerocker** [reply]
ok ive installed v3 by the looks of it.

still problems. it wont allow a search of an entire drive. when prompted i tryped in D:\ and got the error (attached). it then auto-saved the data and created new entries into Engine (attached).

the app shows that it found the tunes, but they weren't in that folder at all. it was only when i pointed towards the correct folder that it found them. however nothing was corrected in Engine and it still has 4 entries for a tune even though its found them. I have no idea what the stuff is below the blue text. but engine isnt updated with finding the tunes.

thanks
Dave
![new bug](https://user-images.githubusercontent.com/84126173/118258896-53e5b300-b4a8-11eb-830f-1c307bfef67b.png)
![engine bug](https://user-images.githubusercontent.com/84126173/118258897-547e4980-b4a8-11eb-9bf2-69dda8ce785c.png)

**Reply by shayded-exe** [reply]
I'll have to fix that error. Looks like I need to filter out system folders like recycle bin. Ideally you should search in a folder, not the root of the drive, but ENJINN should still be able to handle it properly.

**Reply by shayded-exe** [reply]
Also, if you want to undo after running ENJINN, it creates an `m.db.bak` file in your Engine library, just delete `m.db` and rename `m.db.bak` to `m.db`

**Reply by daveceerocker** [reply]
ok, for now i'll just create new tests by wiping Engine before i relocate etc. does the m.db also reset if i start again?

**Comment by daveceerocker** [comment]
ok just for test sake i re-ran the same scenario. this is the trail...

deleted all files from the Engine folder Tunes 1 (remember my engine is on D:)
restarted EngineOS and removed all files (basically new canvas - start again brand new)
Replaced files into Tunes 1 folder in Engine
Added files and created playlist inside Engine - closed Engine - then deleted one file from the Tunes 1 folder in Engine on D:
Engine now shows 1 red file as expected.

I run CMD
used enjinn relocate
app shows missing file as expected
i tell app which folder and its finds the file and says that its Saved Relocated tracks to Engine.
opened EngineOS and file is not saved nor is it there, still showing Red file

hope that helps :)
![same bug](https://user-images.githubusercontent.com/84126173/118261569-a2e11780-b4ab-11eb-85cd-34e45c2104ce.png)

**Reply by shayded-exe** [reply]
Do you have Engine open? It has to be closed when you run ENJINN.

**Reply by daveceerocker** [reply]
i think i did yes...i will try it again but itll be next week as the testing is on my work PC. i'll let you know. (i'll make a video of the test)
thanks

**Reply by shayded-exe** [reply]
Sounds good. I'll try to fix that error you got when saving, or at least have it give a better error message. I'll fix that recycle bin error too.

**Reply by shayded-exe** [reply]
Also to clarify, when you say EngineOS, are you talking about Engine Prime the desktop app, or EngineOS what runs on the smart console?

**Reply by daveceerocker** [reply]
no worries. are you in the UK? 
sadly i thought when it found the files it would actually replace the red ones but it didnt. is that something you can do? i aks because the red files had about 6 hot cues on each track, that was my hope. but this is a good fix anyway.

**Reply by daveceerocker** [reply]
^^^ yeh i mean Engine Prime the actual program not the smart console, sorry for the confusion. i'll refer to it as EngineP from now on.

**Reply by shayded-exe** [reply]
Nah I'm in Texas. And yes it will actually fix the red files. I'll have to upload a video example of it working properly.

In your case, it said it saved, but it didn't actually because of that error right below. That's a bug

**Reply by shayded-exe** [reply]
Ahh ok thought so!

**Reply by daveceerocker** [reply]
yes mate...a video would be awesome!!!

**Reply by shayded-exe** [reply]
Whoop, just found the bug with saving. I'll have a fix out soon.

**Reply by shayded-exe** [reply]
Version [v1.3.1](/rshea0/enjinn/releases/v1.3.1) is out with fixes for both of these issues.

**Comment by daveceerocker** [comment]
Hi, Ok ive now installed this on my main PC.

when asked where the engine folder is giving an error because my C: isnt where it is. this is the path:

D:\

i thought you had fixed this bug. my D drive is small so it not searching 1000's of files.

Now using 1.3.1 (fresh install)

thanks

**Reply by shayded-exe** [reply]
How is your Engine folder not on your C drive? Also, there's no way for your library to be the root of a drive. It will always be in a folder called `Engine Library`.

Engine on Windows always looks for your library at `%USERPROFILE%/Music/Engine Library`.

You want to enter `D:\` when asked where your songs are, not where your Engine folder is.

I just tested relocating songs to an external drive and it worked fine.

![image](https://user-images.githubusercontent.com/1417838/118888807-c0561d00-b8c1-11eb-8441-169abd2e45c7.png)

**Reply by shayded-exe** [reply]
Also, I'm working on a GUI version of the app to make all this easier.

**Comment by daveceerocker** [comment]
My bad!

i was working on my workPC not my home pc. i dont use the engine folder on
my home pc, again i use the D: because theres not enough room on my C:

perhaps you can add that to the program - to allow where to say the root
folder is for engine. (to allow me to input my own path/folder)

i look forward to the GUI version.

**Reply by shayded-exe** [reply]
I'm a bit confused still. Can you please show me your folder structure? You're saying your `m.db` file is at the root of your D drive? How does Engine even access it?

Your engine library folder should not contain your music, just the database files.

**Comment by daveceerocker** [comment]
do you know how far away the GUI version will be please?

i have an idea as well..."Consolidate!" a function that finds all the files
on EnginePrime and actually grabs them and places them in one folder! oh
would that be a dream!

thanks
Dave



On Thu, 20 May 2021 at 18:44, Ryan Shea ***@***.***> wrote:

> I'm a bit confused still. Can you please show me your folder structure?
> You're saying your m.db file is at the root of your D drive? How does
> Engine even access it?
>
> Your engine library folder should not contain your music, just the
> database files.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/enjinn/discussions/22#discussioncomment-764651>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXJ6NNRM2GKRGPVZ7GTTOVDARANCNFSM442GOEXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
A consolidate feature wouldn't be a bad idea.

The GUI version shouldn't take too long, maybe a couple weeks.

**Comment by daveceerocker** [comment]
Thanks, I'll wait for that version.

Is the consolidate function tough to code?

Thanks.

**Reply by shayded-exe** [reply]
Nah, consolidate shouldn't be hard

**Comment by daveceerocker** [comment]
Hi Ryan

I hope you are well mate?

Do you have an update on the new version of Enjinn please?

Ive decided to upgrade to a 3TB SSD. I'll clone my Win10 to that then
expand the C drive so that i can use EnginePrime in its proper place. I
want to dump everything from the D drive to C and the use your incredible
software to auto-locate and hopefully consolidate all the tunes into one
folder. At the moment my entire catalogue is a mess, i have no idea whats
where and tbh its mental crazy!

Cheers Ryan
Dave

On Sun, 23 May 2021 at 05:23, Ryan Shea ***@***.***> wrote:

> Nah, consolidate shouldn't be hard
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/enjinn/discussions/22#discussioncomment-773057>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXPGC5BHOXJKHPJJQE3TPB7LBANCNFSM442GOEXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
I'm getting very close to releasing v2.0 with the GUI. Currently pending on choosing a [new name](https://denondjforum.com/t/made-a-playlist-on-usb-device/32860/10?u=shayded).

I still need to implement library consolidation, but maybe I can squeeze it into v2.0. If not, it will be in v2.1.

Here's a question though. Should consolidation be part of the relocation feature, or should they be separate? Having them separate would be easier and cleaner. Maybe in the initial release they can be separate to make it easier to develop, then I can spend some more time merging the features for a later release if it would make sense. Would give users time to give feedback too.

**Comment by daveceerocker** [comment]
Hi Ryan.

Mate whichever does the job I'm fine with. If it's a separate app then
fine. I just need it to find all files used in engine and then be able to
physically cut or copy those files to a desired folder. Of course all cue
points/data need to travel with the files.

If one program could move files...the other could then relocate them for
engineprime.

This feature would help so many people it's crazy. Coupled with a relocate
that actually works, this is incredible. I'm so envious of your skillz bro.

I have my 2TB SSD arriving tomorrow, can't wait

How long do you think it will be ready?

For a name I'd go with enjinn-saviour! Because that's what it is!!!

Cheers
David

**Reply by shayded-exe** [reply]
Haha I'm really glad you like it so much :)

It'll all be one app, just separate functions. I should have it out within a week.

**Comment by daveceerocker** [comment]
That's fantastic. No really, I'm amazed these features aren't on the
official program. It's on Rekordbox and is a major life saver.

Please can you inform me asap when ready? I'm happy to help with debugging.

Thanks
Dave.

**Reply by shayded-exe** [reply]
[v2.0.0 is out](https://github.com/rshea0/tuneup-prime/releases/tag/v2.0.0) with consolidate included

**Comment by daveceerocker** [comment, marked as answer]
awesome....I'll be installing later.

excellent. :):):):)
